### DISCUSSION TITLE
Testing utilities for named inputs

### DISCUSSION THREAD

**Discussion by kylecannon** [nrwl/nx#27072] [Feature Requests]
As we all know, named inputs are a very important thing to get right in a nx workspace for not only cache hits, but for ensuring that things are being correctly ran. Right now it's very cumbersome to test if the inputs are correct. 

I would love to see the following at minimum:

- Ability to pass in a target and task and get the fully expanded input list, maybe also show the files that are excluded and essentially an explain plan about how the upstream target `a` got merged into `y` target. 
- It would be great to have testing tools to easily validate a cache hit versus a non-cache hit. I see internal tests, but would love to have this be exposed in the nx devkit.

**Comment by rudfoss-rr** [comment]
PLEASE have a look at this one! I've been struggling for weeks with a target that "occasionally" runs instead of using cached values as I expect, and I can't for the life of me figure out why. Some kind of command to show the complete calculation used to determine cache hits and misses would have saved me a lot of time! I'm very curious how the Nx team tests that the cache calculation is working correctly today if there is no such command or mechanism.
